### DISCUSSION TITLE
Querying duckdb through GUI tool?

### DISCUSSION THREAD

**Discussion by AndreasThinks** [duckdb/duckdb#3096] [Q&A] [answered]
All, I've been using DuckbDB for awhile, and it's very nifty...but when I have used DBs previously, I like using gui tools (PyCharm DB, Sql Server Manager, etc) to get auto-complete and such.  Is there a way of opening a persistent database that way? Or am I just being dim?

**Comment by hannes** [comment, marked as answer]
People have reported success using DBeaver with DuckDB

**Reply by AndreasThinks** [reply]
This worked!  And makes me think I can probably use the same principles to import it to PyCharm.  Though interestingly, I had to go one version back.

**Reply by Alex-Monahan** [reply]
Please let us know how things go with pycharm!

**Reply by dominijk** [reply]
@AndreasThinks Were you able to run two connections in paralell i.e. be connected through Python whilst viewing created objects in dbeaver. I'm running into a file lock issue here despite setting dbeaver to be a read only connection and giving python read write, i'm on the latest version (0.6.0). I have a query on the dbeaver repo as : https://github.com/dbeaver/dbeaver/issues/18202

**Reply by Mytherin** [reply]
Supported connection models are a single read-write connection **or** multiple read-only connections, you cannot mix them.

**Reply by alanwilter** [reply]
Why cannot I have e.g. `DBeaver` in read-only mode and my Python app with read-write?

**Reply by Tishj** [reply]
@alanwilter <https://duckdb.org/docs/connect/concurrency.html#handling-concurrency>

**Reply by djouallah** [reply]
> @alanwilter https://duckdb.org/docs/connect/concurrency.html#handling-concurrency

@Tishj just for my own education, is this just a current limitation or it is a fundamental design choice ?

**Reply by Mytherin** [reply]
This is a fundamental design choice. The alternative requires cross-process communication beyond file locks, which is explicitly out-of-scope at least for the core system.

**Reply by djouallah** [reply]
@Mytherin thanks, fwiw, I am very happy with duckdb the way it is :) and will be even more happier when @samansmink delivre delta write 🥇

**Reply by samansmink** [reply]
😅

**Comment by parvizfarnia** [comment]
On windows 10 (64 bits), I have installed DuckDB (Python package) version 0.7.0 and DBeaver version dbeaver-ce-22.3.4-win32.win32.x86_64. I installed the requried library (duckdb_jdbc-0.7.0.jar) for JDBC and I managed to create a connection to my local (persistent) DuckDB database file. However, no matter on what I click, I cannot view any of my tables. Each time I get the following error message

`SQL Error: Actual catalog argument is not supported, got mydb`

(**mydb**.duckdb is the name of my database file)

I even installed several different versions of DBeaver but each time I had the same problem.

Any idea? Am I doing wrong or it's the fact that 0.7.0 is not compatible with DBeaver yet? I had to install this particular version because of its hive-partitioning functionality for Parquet files and I installed its Python version because its R version apparently is not available yet (on CRAN it is written 0.7.0 but the command install.packages installs the version 0.6.2)

**Reply by Mytherin** [reply]
There's an issue getting v0.7.0 to work with DBeaver (#6247) The R version is in the process of being uploaded to CRAN.

**Reply by parvizfarnia** [reply]
Thank you for confirming this. Meantime I will wait for R package in CRAN. Emacs ESS module (for R programming) has a very intersting command : **C-x w** which allows to view table/view contents in Spreadhseet (Excel, Calc, etc.) Of course, it does not replace DBeaver but still provides a good view of the data compared to command line.

**Comment by ryanhamilton** [comment]
I've added and tested DuckDB 0.7 support in qStudio. The driver comes bundled with qStudio download version 2.01:
https://www.timestored.com/qstudio/help/duckdb-sql-editor

If there's enough demand by DuckDB users, I will keep it updated.

![image](https://github.com/duckdb/duckdb/assets/7273208/43d59e6e-5dd1-45a2-8add-0746802a22dc)

**Comment by samansmink** [comment]
FYI DuckDB now has a native UI: https://duckdb.org/2025/03/12/duckdb-ui.html

**Comment by cy-james-lee** [comment]
Jetbrains DataGrip also supports DuckDB through JDBC.
